app/routes/items/new.js

```diff
@@ -69,78 +54,98 @@ export default AuthorizeRoute.extend({
     });
   },
 
-  async setupController(controller, model) {
+  setupController(controller, model) {
     this._super(controller, model);
+    const store = this.get("store");
+    this.initializeController();
+
+    if (this.get("newItemRequest")) {
+      this.initializeDimensions();
+      this.manageSubformDetails();
+      this.setUpPackageImage();
+      window.localStorage.setItem("isSelectLocationPreviousRoute", false);
+    }
+  },
 
-    controller.set("inventoryNumber", this.get("inventoryNumber"));
-    controller.set("displayInventoryOptions", false);
-    controller.set("autoGenerateInventory", true);
-    controller.set("inputInventory", false);
+  initializeController() {
+    const controller = this.controller;
+    if (!controller.get("inventoryNumber")) {
+      controller.send("autoGenerateInventoryNumber");
+    }
     controller.set("invalidLocation", false);
     controller.set("invalidScanResult", false);
     controller.set("labels", 1);
+  },
 
-    if (this.get("newItemRequest")) {
-      this.set("newItemRequest", false);
-      controller.set("quantity", 1);
-      if (
-        window.localStorage.getItem("isSelectLocationPreviousRoute") === "false"
-      ) {
-        controller.set("caseNumber", "");
-        controller.set("length", null);
-        controller.set("width", null);
-        controller.set("height", null);
-        controller.set("selectedGrade", {
-          name: "B",
-          id: "B"
-        });
-        controller.set("imageKeys", "");
-      }
-      let codeId = controller.get("codeId");
-      if (codeId) {
-        let selected = this.get("store").peekRecord("code", codeId);
-        if (selected) {
-          let selectedSubform = selected.get("subform");
-          if (this.isSubformAllowed(selectedSubform)) {
-            controller.set("showAdditionalFields", true);
-            let details = await this.store.query(selectedSubform, {
-              distinct: "brand"
+  async manageSubformDetails() {
+    const controller = this.controller;
+    const store = this.store;
+    let codeId = controller.get("codeId");
+    if (codeId) {
+      let selected = store.peekRecord("code", codeId);
+      if (selected) {
+        let selectedSubform = selected.get("subform");
+        if (this.isSubformAllowed(selectedSubform)) {
+          controller.set("showAdditionalFields", true);
+          let details = await store.query(selectedSubform, {
+            distinct: "brand"
+          });
+          if (this.get("transitionFrom") === "code") {
```

Comment:
We want to avoid those kinds of history lookups. It's not tolerant to change and prevents other usages of the same page.
If there's a specific behaviour the page needs to do, then the caller of the page can pass an argument to it.

Also, why should the page reset values only when it comes from 'code' ? It feels weird